template.yaml

```diff
@@ -10,16 +10,72 @@ Globals:
   Function:
     Timeout: 300
     MemorySize: 600
+    Runtime: python3.9
+
 
 Resources:
+  EqualityCheckerRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Policies:
+        - PolicyName: EqualityCheckerPolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 's3:*'
+                  - 's3-object-lambda:*'
```

Comment:
We can further restrict the S3 access to readonly

Reply:
Would make sense but the equality checker runs a "trusted code" so we're somewhat safe on that side.